contrib/kind-helm.sh

```diff
@@ -0,0 +1,374 @@
+  # bridge interface it creates. This breaks on KIND with IPv4 only deployments, because the new
+  # internal bridge has IPv6 disable and can't move the IPv6 from the eth0 interface.
+  # We can enable IPv6 always in the container, since the docker setup with IPv4 only
+  # is not very common.
+  KIND_NODES=$(kind get nodes --name "${KIND_CLUSTER_NAME}" | grep -v external-load-balancer)
+  for n in $KIND_NODES; do
+    docker exec "$n" sysctl --ignore net.ipv6.conf.all.disable_ipv6=0
+    docker exec "$n" sysctl --ignore net.ipv6.conf.all.forwarding=1
+  done
+}
+
+build_ovn_image() {
+    if [ "${SKIP_OVN_IMAGE_REBUILD}" == "true" ]; then
+      echo "Explicitly instructed not to rebuild ovn image: ${OVN_IMAGE}"
+      return
+    fi
+
+    # Build ovn image
+    pushd ${SCRIPT_DIR}/../go-controller
+    make
+    popd
+
+    # Build ovn kube image
+    pushd ${SCRIPT_DIR}/../dist/images
+    # Find all built executables, but ignore the 'windows' directory if it exists
+    find ../../go-controller/_output/go/bin/ -maxdepth 1 -type f -exec cp -f {} . \;
+    echo "ref: $(git rev-parse  --symbolic-full-name HEAD)  commit: $(git rev-parse  HEAD)" > git_info
+    docker build -t "${OVN_IMAGE}" -f Dockerfile.fedora .
+    popd
+}
+
+get_image() {
+    local image_and_tag="${1:-$OVN_IMAGE}"  # Use $1 if provided, otherwise use $OVN_IMAGE
+    local image="${image_and_tag%%:*}"  # Extract everything before the first colon
+    echo "$image"
+}
+
+get_tag() {
+    local image_and_tag="${1:-$OVN_IMAGE}"  # Use $1 if provided, otherwise use $OVN_IMAGE
+    local tag="${image_and_tag##*:}"  # Extract everything after the last colon
+    echo "$tag"
+}
+
+coredns_patch() {
+  dns_server="8.8.8.8"
+  # No need for ipv6 nameserver for dual stack, it will ask for
+  # A and AAAA records
+  if [ "$IP_FAMILY" == "ipv6" ]; then
+    dns_server="2001:4860:4860::8888"
+  fi
+
+  # Patch CoreDNS to work
+  # 1. Github CI doesn´t offer IPv6 connectivity, so CoreDNS should be configured
+  # to work in an offline environment:
+  # https://github.com/coredns/coredns/issues/2494#issuecomment-457215452
+  # 2. Github CI adds following domains to resolv.conf search field:
+  # .net.
+  # CoreDNS should handle those domains and answer with NXDOMAIN instead of SERVFAIL
+  # otherwise pods stops trying to resolve the domain.
+  # Get the current config
+  original_coredns=$(kubectl get -oyaml -n=kube-system configmap/coredns)
+  echo "Original CoreDNS config:"
+  echo "${original_coredns}"
+  # Patch it
+  fixed_coredns=$(
+    printf '%s' "${original_coredns}" | sed \
+      -e 's/^.*kubernetes cluster\.local/& net/' \
+      -e '/^.*upstream$/d' \
+      -e '/^.*fallthrough.*$/d' \
+      -e 's/^\(.*forward \.\).*$/\1 '"$dns_server"' {/' \
+      -e '/^.*loop$/d' \
+  )
+  echo "Patched CoreDNS config:"
+  echo "${fixed_coredns}"
+  printf '%s' "${fixed_coredns}" | kubectl apply -f -
+}
+
+create_kind_cluster() {
```

Comment:
shouldn't we try to consolidate these functions into a common bash library file and then source it it in kind.sh and kind-helm.sh...so that we have a common library and are not copying code?

Reply:
yes! In fact, the install metal-lb is one of these functions that would move to that common bash file, @trozet .
Would you be okay with me doing that as part of https://github.com/ovn-org/ovn-kubernetes/issues/4257 ?